src/cloud_provider/aws/kubernetes/mod.rs

```diff
@@ -119,1675 +109,1168 @@ pub struct Options {
+            let deletion = cmd::kubectl::kubectl_exec_delete_namespace(
+                &kubernetes_config_file_path,
+                qovery_namespace,
+                kubernetes.cloud_provider().credentials_environment_variables(),
+            );
+            match deletion {
+                Ok(_) => kubernetes.logger().log(EngineEvent::Info(
+                    event_details.clone(),
+                    EventMessage::new_from_safe(format!("Namespace {} is fully deleted", qovery_namespace)),
+                )),
+                Err(e) => {
+                    if !(e.message(ErrorMessageVerbosity::FullDetails).contains("not found")) {
+                        kubernetes.logger().log(EngineEvent::Warning(
+                            event_details.clone(),
+                            EventMessage::new_from_safe(format!("Can't delete namespace {}.", qovery_namespace)),
+                        ))
+                    }
+                }
+            }
         }
 
-        self.send_to_customer(
-            format!("Upgrading Kubernetes {} worker nodes", self.name()).as_str(),
-            &listeners_helper,
-        );
-        self.logger().log(EngineEvent::Info(
+        kubernetes.logger().log(EngineEvent::Info(
             event_details.clone(),
-            EventMessage::new_from_safe("Upgrading Kubernetes worker nodes.".to_string()),
+            EventMessage::new_from_safe("Delete all remaining deployed helm applications".to_string()),
         ));
 
-        // Disable cluster autoscaler deployment
-        let _ = self.set_cluster_autoscaler_replicas(event_details.clone(), 0)?;
-
-        match terraform_init_validate_plan_apply(temp_dir.as_str(), self.context.is_dry_run_deploy()) {
-            Ok(_) => {
-                self.send_to_customer(
-                    format!("Kubernetes {} workers nodes have been successfully upgraded", self.name()).as_str(),
-                    &listeners_helper,
-                );
-                self.logger().log(EngineEvent::Info(
-                    event_details.clone(),
-                    EventMessage::new_from_safe(
-                        "Kubernetes workers nodes have been successfully upgraded.".to_string(),
-                    ),
-                ));
+        match helm.list_release(None, &[]) {
+            Ok(helm_charts) => {
+                for chart in helm_charts {
+                    let chart_info = ChartInfo::new_from_release_name(&chart.name, &chart.namespace);
+                    match helm.uninstall(&chart_info, &[]) {
+                        Ok(_) => kubernetes.logger().log(EngineEvent::Info(
+                            event_details.clone(),
+                            EventMessage::new_from_safe(format!("Chart `{}` deleted", chart.name)),
+                        )),
+                        Err(e) => {
+                            let message_safe = format!("Error deleting chart `{}`: {}", chart.name, e);
```

Comment:
Same, potential leak here, do not fill error message into safe message.

```
kubernetes.logger().log(EngineEvent::Warning(
                                event_details.clone(),
                                EventMessage::new(format!("Error deleting chart `{}`", chart.name), Some(e.to_string())),
                            ))
```

Reply:
I just refactored the existing code base. It was already present

Reply:
Done, fixed into AWS here => #700